Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/WeaponData.xml

```diff
@@ -3071,4 +3071,45 @@
         <AllowedMovement value="Moving"/>
         <TeleportResetRange value="0"/>
     </CWeaponLegacy>
+    <CWeaponLegacy id="AP_VoidRayPurifierBeamLaunch">
+        <EditorCategories value="Race:Protoss"/>
+        <Options index="Hidden" value="1"/>
+        <Icon value="Assets\Textures\btn-upgrade-protoss-airweaponslevel0.dds"/>
+        <DisplayEffect value="AP_VoidRayPurifierBeamDamage"/>
+        <TargetFilters value="Visible,Ground;Missile,Stasis,Dead,Hidden,Invulnerable"/>
+        <Range value="6"/>
+        <RangeSlop value="2"/>
+        <Period value="1.5"/>
+        <Backswing value="0.75"/>
+        <DamagePoint value="0"/>
+        <Effect value="AP_VoidRayPurifierBeamLaunchPersistent"/>
+        <AllowedMovement value="Moving"/>
+        <TeleportResetRange value="0"/>
+        <Options index="OnlyFireWhileInAttackOrder" value="0"/>
+        <Options index="OnlyFireAtAttackOrderTarget" value="0"/>
+        <LegacyOptions index="CanRetargetWhileChanneling" value="1"/>
+    </CWeaponLegacy>
+    <CWeaponStrafe id="AP_VoidRayPurifierBeamDummyWeapon">
+        <Effect value="AP_PointDefenseLaserDummy"/>
+        <EditorCategories value="Race:Protoss"/>
+        <Options index="Hidden" value="1"/>
+        <AcquireCallForHelpFilters value="-;Player,Ally,Neutral,Enemy"/>
+        <TargetFilters value="Visible,Ground;Missile,Stasis,Dead,Hidden,Invulnerable"/>
+        <Range value="1"/>
+<!--        loiter radius = "area of effect" for this weapon-->
+        <LoiterRadius value="2"/>
+        <TeleportResetRange value="0"/>
+        <Arc value="10"/>
+        <Period value="5"/>
+    </CWeaponStrafe>
+    <CWeaponLegacy id="AP_VoidRayPurifierWeaponInfo">
+        <EditorCategories value="Race:Protoss"/>
+        <Icon value="Assets\Textures\btn-upgrade-protoss-airweaponslevel0.dds"/>
+        <DisplayEffect value="AP_VoidRayPurifierBeamDamage"/>
+        <TargetFilters value="Visible;Player,Ally,Neutral,Enemy"/>
+        <Range value="6"/>
+        <Arc value="360"/>
+        <Period value="0.2"/>

```

Comment:
Check the displayed attack speed (for beams it should match Effect's `AP_VoidRayPurifierBeamLaunchPersistent` `PeriodicPeriodArray` value)

Reply:
The `VoidRayPurifierBeamLaunchPersistent` effect determines the cooldown between uses of the "magazine", from my testing; it's a relic of the Carrier Interceptor functionality that I used as a baseline, and it doesn't have much impact unless it has a longer period than the `VoidRayPurifierBeamLaunch` weapon. 

The displayed attack speed should (and currently does) match the period of the `VoidRayPurifierBeamDamageField` behavior on the beams themselves, which is responsible for dealing the damaging effect.